readme.txt
[hr]
[center][color=red][size=16pt][b]POST & PM INLINE ATTACHMENTS v6.4[/b][/size][/color]
[url=http://www.simplemachines.org/community/index.php?action=profile;u=253913][b]By Dougiefresh[/b][/url] -> [url=http://custom.simplemachines.org/mods/index.php?mod=3770]Link to Mod[/url]
[/center]
[hr]

[color=blue][b][size=12pt][u]Introduction[/u][/size][/b][/color]
This mod adds the ability to position your attachments in either your forum post or your personal message post using [attachment=n][/attachment] bbcode (where [b]n[/b] is the number of the attachment in the post, eg first = 0, second = 1).

[b]NOTICE:[/b]  In order to support inline attachments in personal messages (PMs), you are [b]REQUIRED[/b] to have the [url=https://custom.simplemachines.org/mods/index.php?mod=1974]PM Attachments[/url] mod [b]VERSION 2.6 OR BETTER[/B] installed, prior to this installation of this mod!!

[color=blue][b][size=12pt][u]New BBcodes[/u][/size][/b][/color]
This mod supports 5 new bbcodes in order to position your attachments inline:
o [b]attachment[/b] => Show full expanded picture
o [b]attach[/b] => Show thumbnail, expandable to full picture
o [b]attachthumb[/b] => Show thumbnail ONLY, not expandable
o [b]attachmini[/b] => Show thumbnail, expandable to full picture
o [b]attachurl[/b] => Shows attachment size, iamge dimensions, and download count; no picture

[color=blue][b][size=12pt][u]BBcodes Parameters[/u][/size][/b][/color]
Each new BBCode accepts the following formats:
[code=Version 1.x]
[nobbc][attach=[/nobbc][b]{id}[/b]][/attach]
[nobbc][attach=[/nobbc][b]{id}[/b],[b]{width}[/b]][/attach]
[nobbc][attach=[/nobbc][b]{id}[/b],[b]{width}[/b],[b]{height}[/b]][/attach]
[/code]
In each case, [b]{id}[/b] is the attachment number relative to the topic, [b]{width}[/b] is the max desired width, [b]{height}[/b] is the max desired height, [b]{pixels}[/b] is the number of pixels surrounding the image, and [b]{float}[/b] can be either [b]left[/b], [b]right[/b], or [b]center[/b].  All text between the opening and closing attachment tags is discarded.

[color=blue][b][size=12pt][u]Version 2.0+ BBcode Parameters[/u][/size][/b][/color]
Version 2.0+ introduced a new format that allows the following parameters:
[code=Version 2.x]
[nobbc][attachment[/nobbc] id=[b]{id}[/b] {parameter}={value}][/attachment]
[/code]
Allowed parameters:
[table]
[tr][td]id[/td][td]{attachment id}[/td][td]ID number of the attachment to show inline (NOT attachment number!)[/td][/tr]
[tr][td]width[/td][td]{width}[/td][td]Desired width of image to show.  Valid: positive integers.[/td][/tr]
[tr][td]height[/td][td]{height}[/td][td]Desired height of image to show.  Valid: positive integers.[/td][/tr]
[tr][td]float[/td][td]{float}[/td][td]Floats image to relation to everything else.  Valid: left, right, center[/td][/tr]
[tr][td]margin[/td][td]{pixels}[/td][td]Margin around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]margin-left[/td][td]{pixels}[/td][td]Left margin around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]margin-right[/td][td]{pixels}[/td][td]Right margin around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]margin-top[/td][td]{pixels}[/td][td]Top margin around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]margin-bottom[/td][td]{pixels}[/td][td]Bottom margin around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]border-style[/td][td]{style}[/td][td]Border style.  Valid: none, dotted, dashed, solid, double, groove, ridge, inset, outset[/td][/tr]
[tr][td]border-width[/td][td]{pixels}[/td][td]Border width around inline attachment.  Valid: positive integers[/td][/tr]
[tr][td]border-color[/td][td]{color}[/td][td]Border color.  Valid formats: plain text, #xxx, #xxxxxx, rbg(d,d,d)[/td][/tr]
[tr][td]scale[/td][td]{answer}[/td][td]Override scaling of image.  Valid: true, false, yes, no[/td][/tr]
[tr][td]msg[/td][td]{msg ID}[/td][td]Message ID number.  Valid: positive integers.[/td][/tr]
[/table]
In each case, [b]{id}[/b] is the attachment number relative to the topic, [b]{width}[/b] is the max desired width, [b]{height}[/b] is the max desired height, [b]{pixels}[/b] is the number of pixels surrounding the image, and [b]{float}[/b] can be either [b]left[/b], [b]right[/b], or [b]center[/b].  All text between the opening and closing attachment tags is discarded.

[color=blue][b][size=12pt][u]Version 3.0+ BBcode Format[/u][/size][/b][/color]
[b]Version 3.0[/b] makes further changes and allows the use of the inline attachments bbcodes [b]WITHOUT[/b] closing brackets, as well as using attachments from another post!

[b]Version 3.11[/b] makes further changes and allows the use of closed tags, like [b][nobbc][attach][/nobbc][/b].  This new form is autonumbered!  Note that the [b][nobbc][attach][/nobbc][/b] tag is processed first, then [b][nobbc][attachment][/nobbc][/b], then [b][nobbc][attachmini][/nobbc][/b], then [b][nobbc][attachthumb][/nobbc][/b], then [b][nobbc][attachurl][/nobbc][/b].

[color=blue][b][size=12pt][u]Other Mod Features[/u][/size][/b][/color]
o Error Text strings are shown for invalid/missing/deleted attachments.
o Inline attachment processes takes place in the [b]parse_bbc[/b] function, which means any parsing requests can benefit from this mod!
o Text string is shown as alternative in code
o Adds [Insert Attachment x] next to each attachment/upload box to insert the bbcode.
o Attachments used by the inline attachments mod can be omitted from the attachment display at the bottom of the post
o Reloads the attachments for Ajax Editing.
o Removing an attachment removes the attachment bbcode for that attachment & changes remaining attachment tags to ensure proper post appearance.
o Text between inline attachment brackets are removed (as of version 3.0).
o Automatic modification of boilerplates that use the ILA tags within them when changing "1-based" numbering option.

[color=blue][b][size=12pt][u]Admin Settings[/u][/size][/b][/color]
The bbcode may be disabled by going to [b][i]Admin[/i] -> [i]Forums[/i] -> [i]Posts and Topics[/i] -> [i]Bulletin Board Code[/i][/b] and unchecking the [b]attachment[/b] bbcode.

On the [b][i]Admin[/i] -> [i]Layout[/i] -> [i]Attachments and Avatars[/i] -> [i]Inline Attachments[/i][/b] page, there are several new options:
o Remove attachment image under post after in-post use.
o Use "One based numbering" for attachment IDs (first attachment is 1 instead of 0).
o Allow quoted attachment images from another post.
o Show download link and counter under inline attachment, like non-inline attachments.
o Turn off "nosniff" option for IE and Chrome browsers.
o Whether "attach" tag is the same as the "attachment" tag.
o Allow playing inline attachments that are videos.
o Use Highslide effects for inline attachments. (only if supported Highslide mod is installed)
o Show EXIF information (only if [url=http://custom.simplemachines.org/mods/index.php?mod=169]EXIF[/url] mod is installed)

[color=blue][b][size=12pt][u]Compatibility Notes[/u][/size][/b][/color]
This mod was tested on SMF 2.0.13, but should work on SMF 2.1 Beta 3, as well as SMF 2.0 and up.  SMF 2.1 Beta 1, SMF 2.1 Beta 2, and SMF 1.x will not be supported.

For SMF 2.1 Beta 3, this mod contains no functionality for PM attachments, and posting regular attachments has been changed slightly to allow only 1 file per input box.

These mods can be installed at any time (not required):
o [url=http://custom.simplemachines.org/mods/index.php?mod=1605]JQLightBox[/url]
o [url=http://custom.simplemachines.org/mods/index.php?mod=3594]SCEditor4Smf[/url]
o [url=http://custom.simplemachines.org/mods/index.php?mod=169]EXIF[/url]
o [url=http://custom.simplemachines.org/mods/index.php?mod=2233]Boilerplates for Posts[/url]

These mods should be installed before this mod (not required):
o [url=http://custom.simplemachines.org/mods/index.php?mod=1450]Highslide Image Viewer[/url]
o [url=https://github.com/Spuds/SMF-HS4SMF]HS4SMF v0.8.1[/url]
o [url=http://custom.simplemachines.org/mods/index.php?mod=1974]PM Attachments[/url]
o [url=http://custom.simplemachines.org/mods/index.php?mod=2758]Custom View of Attachments[/url]

[color=blue][b][size=12pt][u]Changelog[/u][/size][/b][/color]
The changelog has been removed and can be seen at [url=http://www.xptsp.com/board/index.php?topic=12.msg137#msg137]XPtsp.com[/url].

[color=blue][b][size=12pt][u]License[/u][/size][/b][/color]
[quote]Copyright (c) 2013 - 2018, Douglas Orend
All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
[/quote]

Changelog.txt
v6.4 - February 17th, 2018
o Implemented [url=https://www.simplemachines.org/community/index.php?action=profile;u=112942]GL700Wing[/url]'s fixes, as shown [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3962266#msg3962266]here[/url].

v6.3 - January 7th, 2018
o Fixed coding syntax error in [b]Subs-InlineAttachments.php[/b].

v6.2 - December 18th, 2017
o Fixed errors generated by NULL being was passed to [b]in_array[/b] function.
o Fixed code fragment using [b]ILA_mime_type[/b] in [b]Display.php[/b].

v6.1 - December 15th, 2017
o Fixed installer where ILA actions are inserted.

v6.0 - December 8th, 2017
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950088#msg3950088]Per GL700Wing:[/url] Fixed issue where inserting a PM attachment inline doesn't work.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950088#msg3950088]Per GL700Wing:[/url] Fixed issue where inline PM attachments were being displayed as unapproved.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950088#msg3950088]Per GL700Wing:[/url] Fixed install issue with PM Attachments mod regarding [b]ManageAttachments.php[/b].
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950088#msg3950088]Per GL700Wing:[/url] Fixed issue where inline PM attachments are shown again under PM text.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950088#msg3950088]Per GL700Wing:[/url] Fixed issue where inline PM attachments can't be downloaded.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950101#msg3950101]Per GL700Wing:[/url] Fixed issue where no option to clear/remove the first attachment when composing a new message.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950109#msg3950109]Per GL700Wing:[/url] Fixed issue where [nobbc][attachthumb][/nobbc] is an expandable thumbnail.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950802#msg3950802]Per GL700Wing:[/url] Fixed issue so that inline attachments can be quoted in Tapatalk.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3950933#msg3950933]Per GL700Wing:[/url] Fixed another issue code so that inline attachments can be quoted in Tapatalk.
o [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3951094#msg3951094]Per GL700Wing:[/url] Fixed issue with Custom Attachment View mod where all attachments are shown.
o Copied mime detection code from my [url=http://custom.simplemachines.org/mods/index.php?mod=4131]Play Media Attachments[/url] mod.

v5.13 - November 26th, 2017
o Added support for the Tapatalk mod.
o Fixed undefined variable error when HS4SMF mod is installed and enabled.

v5.12 - November 14th, 2017
o Modified code to enclose ILA tag with "display: inline-block" CSS style, based on [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3945482#msg3945482]@rjen's code snipet.[/url].
o Fixed attachment not inline after editing post, per [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3947759#msg3947759]GL700Wing's instructions[/url].
o Fixed download link/counter code for display, per [url=https://www.simplemachines.org/community/index.php?topic=525705.msg3947786#msg3947786]GL700Wing's instructions[/url].

v5.11 - October 18th, 2017
o Modified code to return already-build attachment context array if available.
o Fixed reporting invalid attachment when viewing attachments on board wasn't possible.

v5.10 - September 8th, 2017
o Fixed the HTML building code so various parameters work properly.
o Fixed undefined array element error in [b]Subs-InlineAttachments.php[/b].

v5.9 - August 31st, 2017
o Completed SMF 2.1 Beta 3 template edits.  Javascript edits still needed.
o Replaced IFNULL with COALESCE in database queries for better Postgres support.

v5.8 - July 8th, 2017
o Updated source code for SMF 2.1 Beta 1 & 2.  Templates are still unedited.
o Remvoed support for SMF 2.1 Beta 1 & 2.
o Attempt fix for issue where incomplete information was pulled for attachments.

v5.7 - July 5th, 2017
o Fixed 2 undefined index errors found by [url=http://www.simplemachines.org/community/index.php?action=profile;u=112942]GL700Wing[/url].  Thanks, GL700Wing!
o Resolved undefined array element errors coming from [url=http://custom.simplemachines.org/mods/index.php?mod=4087]Automatic Attachment Rotation[/url] mod.  Thanks, [url=http://www.simplemachines.org/community/index.php?action=profile;u=112942]GL700Wing[/url]!!

v5.6 - March 15th, 2017
o Fixed regex expressions so that [b]code[/b] & [b]quote[/b] tags are processed correctly.

v5.5 - March 11th, 2017
o Fixed "(more attachments)" issue.  Thanks, [url=http://www.simplemachines.org/community/index.php?action=profile;u=52327]LG965[/url]!

v5.4 - March 10th, 2017
o Fixed readme compatibility note about HS4SMF mod being installed first.
o Fixed issue where topic ID was changed by code pulling attachments.
o Fixed issue where ILA tags within code tags were incorrectly translated.
o Fixed issue with Highslide mods and maximum width/height settings.
o Fixed incompatibility with [url=http://custom.simplemachines.org/mods/index.php?mod=3832]URL Popup[/url] mod.
o Fixed error where unable to load language file after uninstalling.

v5.3.1 - March 5th, 2017
o Removed debug statement accidentally left in the mod.

v5.3 - March 4th, 2017
o Fixed code that returns the image dimensions to the HTML building code.
o Removed ILA popup window with ILA parameters.  Will be replaced by new stuff....

v5.2 - February 24th, 2017
o Fixed issue where popup option wasn't linked to anything....
o Fixed responsive CSS image stuff that I added in version 5.0.
o Fixed [b]scale[/b] parameter to disable/enable responsive stuff on request.
o Added admin option to enable responsive CSS image stuff.

InlineAttachmentsAdmin.english.php
<?php
/**********************************************************************************
* InlineAttachmentsAdmin.english.php - English language file
***********************************************************************************
* This mod is licensed under the 2-clause BSD License, which can be found here:
*	http://opensource.org/licenses/BSD-2-Clause
***********************************************************************************
* This program is distributed in the hope that it is and will be useful, but      *
* WITHOUT ANY WARRANTIES; without even any implied warranty of MERCHANTABILITY    *
* or FITNESS FOR A PARTICULAR PURPOSE.                                            *
**********************************************************************************/
if (!defined('SMF')) 
	die('Hacking attempt...');

global $helptxt;

// Regular language strings:
$txt['ila_admin_settings'] = 'Inline Attachments';
$txt['ila_new_version'] = 'Post and PM Inline Attachments mod version %s is available for download!';
$txt['ila_no_update'] = 'Your install of Post and PM Inline Attachments is up to date!';
$txt['ila_completed_singular'] = '1 post with inline attachment tags have been updated!';
$txt['ila_completed_plural'] = '%d posts with inline attachment tags have been updated!';

$txt['ila_title'] = 'Post and PM Inline Attachments';
$txt['ila_insert_tag'] = 'Tag to use when inserting inline attachments:';
$txt['ila_attach_same_as_attachment'] = '&quot;attach&quot; bbcode same as &quot;attachment&quot;?';
$txt['ila_insert_format'] = 'Format to insert new attachment tags:';

$txt['ila_max_width'] = 'Restrict images to maximum width of:<div class="smalltext">NOTE: <strong>0</strong> = disabled</div>';
$txt['ila_max_height'] = 'Restrict images to maximum height of:<div class="smalltext">NOTE: <strong>0</strong> = disabled</div>';
$txt['ila_enable_responsive'] = 'Enable responsive CSS for inline attachments?';

$txt['ila_highslide'] = 'Enable Highslide effects for inline attachment?';
$txt['ila_one_based_numbering'] = 'Should first attachment should be numbered &quot;1&quot;?';
$txt['ila_one_based_numbering_ask'] = 'Do you want to adjust all inline attachment IDs so that they still work the same as before?  Press \"OK\" to do this upon saving.';
$txt['ila_allow_quoted_images'] = 'Allow quoted attachment images from another post?';
$txt['ila_duplicate'] = 'Remove attachment image under post after in-post use?';
$txt['ila_download_count'] = 'Download link and counter setting:';
$txt['ila_download_count_n'] = 'Disabled';
$txt['ila_download_count_f'] = 'Filename only';
$txt['ila_download_count_fs'] = 'Filename and Size';
$txt['ila_download_count_fsd'] = 'Filename, Size and Dimensions';
$txt['ila_download_count_fsdc'] = 'Filename, Size, Dimensions and Count';
$txt['ila_download_count_fsdc2'] = 'Filename, Size, Dimensions and Count (2 Lines)';
$txt['ila_download_count_fsdc3'] = 'Filename, Size, Dimensions and Count (3 Lines)';
$txt['ila_transparent'] = 'Unapproved images shown as transparent: (%)<div class="smalltext">NOTE: <strong>0</strong> disables showing unapproved images</div>';

//$txt['ila_popup_help'] = 'Include link for ILA popup window parameters?';
//$txt['ila_no_popup_on_new'] = 'Include popup for new posts/PMs?';

$txt['ila_embed_video_files'] = 'Embed supported video formats within the post?';
$txt['ila_video_default_width'] = 'Default width of video when width isn\'t specified:';
$txt['ila_video_default_height'] = 'Default height of video when height isn\'t specified:';
$txt['ila_video_show_download_link'] = 'Show download link for videos?';
$txt['ila_video_html5'] = 'Check to use HTML5 video tags to show video:';

$txt['ila_embed_svg_files'] = 'Embed SVG files within the post?';
$txt['ila_embed_txt_files'] = 'Embed TXT files as part of the post?';
$txt['ila_embed_pdf_files'] = 'Embed PDF files within the post?';

$txt['ila_turn_nosniff_off'] = 'Turns &quot;nosniff&quot; option off for IE 8+?';
$txt['ila_display_exif'] = 'Display EXIF information below image?';

// Help language strings:
$helptxt['ila_insert_tag'] = 'This option allows you to choose which inline attachment tag to use with the &quot;Insert Attachment&quot; link after each uploaded attachment.<br /><br />
	Your choices are:
	<blockquote class="bbc_standard_quote">
		<strong>attachment</strong> - Show attachment as full size image<br />
		<strong>attach</strong> - Shows the attachment as a thumbnail, expandable to full size.<br />
		<strong>attachthumb</strong> - Shows NON-EXPANDABLE thumbnail of the attachment.<br />
		<strong>attachmini</strong> - Shows the attachment as a thumbnail, expandable to full size.  Download count and filename omitted beneath (regardless of ILA settings).
		<strong>attachurl</strong> => Shows the attachment like you used an [b]url[/b] tag instead of this tag.
	</blockquote>
	The default inline attachment tag is <strong>attachment</strong>.';
$helptxt['ila_attach_same_as_attachment'] = 'When checked, this option changes the <strong>attachment</strong> tag so that it shows the attachment as a thumbnail, expandable to full size, like the <strong>attach</strong> tag does.<br /><br />The default setting (<strong>unchecked</strong>) is to display the attachment as full size.';
$helptxt['ila_highslide'] = 'When checked, this option uses one of the supported mods to display the image using <a href="http://www.highslide.com/">Highslide</a> effects.<br /><br />The following mods are supported:
	<blockquote class="bbc_standard_quote">
		<strong><a href="https://github.com/Spuds/SMF-HS4SMF">HS4SMF v0.8.1</a></strong> (required to be installed prior to ILA!)<br />
		<strong><a href="http://custom.simplemachines.org/mods/index.php?mod=1450">Highslide Image Viewer</a></strong><br />
		<strong><a href="http://custom.simplemachines.org/mods/index.php?mod=1605">JQLightBox</a></strong>
	</blockquote>';
$helptxt['ila_one_based_numbering'] = 'When checked, this option changes all inline attachments so that the first attachment is numbered <strong>1</strong> (one), as opposed to the default <strong>0</strong> (zero).<br /><br />Note that changing this option will cause the mod to alter all inline attachment tags throughout your forum so that this option will correctly display with the changed setting.';
$helptxt['ila_allow_quoted_images'] = 'When checked and without the <a href="https://tapatalk.com/download_SimpleMachines.php">Tapatalk</a> SMF plugin, you can quote images and have the inline attachment show up correctly.  Otherwise, the inline attachment is replaced with a placeholder text string indicating where the inline attachment was in the quoted post.<br /><br />With the <a href="https://tapatalk.com/download_SimpleMachines.php">Tapatalk</a> SMF plugin installed, this option is not available for use, as it breaks the Tapatalk app for some reason.  Future versions of this mod may resolve this issue.';
$helptxt['ila_duplicate'] = 'When checked, this option removes the attachment from the attachment list after the post.<br /><br />Default is <strong>checked</strong>.';
$helptxt['ila_download_count'] = 'This option controls what is displayed beneath the inline attachment.  The following options are present:
	<blockquote class="bbc_standard_quote">
		<strong>Disabled</strong> - No text is displayed beneath.<br />
		<strong>Filename Only</strong><br />
		<strong>Filename and Size</strong><br />
		<strong>Filename, Size, and Dimensions</strong><br />
		<strong>Filename, Size, Dimensions and Count</strong><br />
		<strong>Filename, Size, Dimensions and Count (2 lines)</strong><br />
		<strong>Filename, Size, Dimensions and Count (3 lines)</strong><br />
	</blockquote><br />
	With the <strong>Filename, Size, Dimensions and Count</strong> option, everything is display on one line, like this:
	<blockquote class="bbc_standard_quote">
		<img src="Themes/default/images/icons/clip.gif" alt="*" align="middle" border="0"></img>Desert.jpg (826.11 kB . 1024x768 - viewed 5 times) ::)
	</blockquote><br />
	With the 2-line option, download link is on the first line; Size, dimensions and download count are on the second line.  Example:
	<blockquote class="bbc_standard_quote">
		<img src="Themes/default/images/icons/clip.gif" alt="*" align="middle" border="0"></img>Desert.jpg<br />
		(826.11 kB . 1024x768 - viewed 3 times) ::)
	</blockquote><br />
	With the 3-line option, download link is on the first line; Size and dimensions are on the second line; Download count is on the third line.
	<blockquote class="bbc_standard_quote">
		<img src="Themes/default/images/icons/clip.gif" alt="*" align="middle" border="0"></img>Desert.jpg<br />
		(826.11 kB . 1024x768)<br />
		(viewed 2 times)<br />
	</blockquote>';
$helptxt['ila_transparent'] = 'This option changes how transparent unapproved images are, as a percentage from <strong>0</strong> to <strong>100</strong>.  Setting this value to <strong>0</strong> disables this option.  Members who can\'t see unapproved images aren\'t effected by this option.<br /><br />Default value is <strong>40</strong>, as in <strong>40%</strong>.';
$helptxt['ila_embed_video_files'] = 'When checked, this option allows you to embed the following video files into a post:
	<blockquote class="bbc_standard_quote">
		<strong>AVI</strong> - played using <strong><a href="http://www.divx.com">DivX</a></strong>\'s player.  Assumes the DivX codec is installed on the OS being browsed.<br />
		<strong>WMV</strong> - played using Windows Media Player.<br />
		<strong>MP4</strong> - played using HTML5 tags (when &quot;Video HTML5&quot; is checked), with fallback HTML provided by <a href="http://www.jwplayer.com">JWPlayer</a>\'s Flash player.<br />
		<strong>WebM</strong> - played using HTML5 tags (when &quot;Video HTML5&quot; is checked), with fallback HTML provided by <a href="http://www.jwplayer.com">JWPlayer</a>\'s Flash player.<br />
		<strong>OGV</strong> - played using HTML5 tags (when &quot;Video HTML5&quot; is checked), with no fallback HTML provided.<br />
	</blockquote>';
$helptxt['ila_video_default_width'] = 'This option specifies the default display width of the inline video attachment.  It can be overriden by using the <strong>width</strong> parameters in the inline attachment tag usage, such as:<code class="bbc_code">[attachment id=x width=640]</code><br/>Default value is <strong>640</strong>.';
$helptxt['ila_video_default_height'] = 'This option specifies the default display height of the inline video attachment.  It can be overriden by using the <strong>height</strong> parameters in the inline attachment tag usage, such as:<code class="bbc_code">[attachment id=x height=640]</code><br/>Default value is <strong>400</strong>.';
$helptxt['ila_video_show_download_link'] = 'When checked, this option provides a download link to the user so that if the video doesn\'t play on the website, the user can download the video in order to play it locally.';
$helptxt['ila_video_html5'] = 'When checked, this option sends HTML5 tags in order to attempt to play <strong>MP4</strong>, <strong>ogv</strong>, and <strong>WebM</strong> file formats.  Regardless of this setting, fallback HTML code is provided using <a href="http://www.jwplayer.com">JWPlayer</a>\'s Flash player.<br /><br /><strong>RESTRICTIONS:</strong> In order to successfully play video files using HTML5 without using the fallback HTML, the poster is required to upload video files in all <strong>3</strong> formats.  If all formats are not provided and the video HTML tags cannot play the format for the browser, the fallback HTML code will be used to attempt to play the video file.  Note that the video file cannot reside on a <strong>localhost</strong>, <strong>127.0.0.1</strong>, or internal network (<strong>192.168.x.x</strong>) IP address.';
$helptxt['ila_embed_svg_files'] = 'When checked, this option allows you to embed SVG (Scalable Vector Graphics) images into your post, like a regular <strong>JPG</strong>, <strong>GIF</strong>, or <strong>PNG</strong> inline attachment.  Otherwise, inline SVG attachments cannot be displayed normally.<br /><br />Default is <strong>unchecked</strong>.';
$helptxt['ila_embed_txt_files'] = 'When checked, this option allows you to embed the contents of a TXT file into the post.<br /><br />Default is <strong>unchecked</strong>.';
$helptxt['ila_embed_pdf_files'] = 'When checked, this option allows you to embed an attached PDF file into the post using the Google Docs service.  Note that Google Docs cannot pull the attachment from <strong>localhost</strong> and <strong>127.0.0.1</strong> addresses, nor unresolvable IP addresses (most notably network IPs behind a router, for example: <strong>192.168.1.1</strong>).<br /><br />Default is <strong>unchecked</strong>.';
$helptxt['ila_turn_nosniff_off'] = 'Checking this option may help to resolve issues where the MIME type of an attachment doesn\'t match the actual type of the attachment and the users are using IE8+.<br /><br />
	<strong>What the &quot;nosniff&quot; option does in IE 8+:</strong>
	<blockquote class="bbc_standard_quote">
		<strong>MIME-handling changes</strong>. The following changes are made to the Internet Explorer 8 Multipurpose Internet Mail Extensions (MIME)-type detection algorithms:<br /><br />
		<strong>Restrict MIME-type detection</strong>. Internet Explorer 8 prevents the detection, or data sniffing, of files with image/* MIME content types into HTML or script. If a file contains script and the server declares that it is an image file, Internet Explorer 8 does not run the embedded script.<br /><br />
		<strong>Prevent MIME-type detection</strong>. Web applications now can prevent MIME-type detection. Sending the new X-Content-Type-Options: nosniff header prevents Internet Explorer from using MIME-type detection to change the server-declared content type.<br /><br />
		<strong>Force save</strong>. For Web applications that need to serve untrusted HTML files, Internet Explorer 8 contains a mechanism that forces users to save untrusted HTML files locally before opening to help prevent the untrusted content from compromising site security.<br /><br />
	</blockquote>
	Source: <a href="https://technet.microsoft.com/en-us/library/dd919181%28v=WS.10%29.aspx">Microsoft TechNet article</a>';
$helptxt['ila_display_exif'] = 'When used in conjunction with the <a href="http://custom.simplemachines.org/mods/index.php?mod=169">EXIF</a> mod, this option allows you to display any EXIF information that the mod can pull from the image.<br /><br />
	<strong>What EXIF information is:</strong>
	<blockquote class="bbc_standard_quote">
		EXIF is short for Exchangeable Image File, a format that is a standard for storing interchange information in digital photography image files using JPEG compression. Almost all new digital cameras use the EXIF annotation, storing information on the image such as shutter speed, exposure compensation, F number, what metering system was used, if a flash was used, ISO number, date and time the image was taken, whitebalance, auxiliary lenses that were used and resolution. Some images may even store GPS information so you can easily see where the images were taken!
	</blockquote>
	Source: <a href="www.exifdata.com">EXIFdata.com</a>';
$helptxt['ila_enable_responsive'] = 'Checking this option dynamically resizes the images so that they stay inside the post area of the forum, regardless of the size of the window.  Not having this option checked and not having a specific width/height defined will result in images that spill over the post area and result in the need to use the scrollbars to view the entire image.';

?>